lib/measures/CreateTypicalBuilding/measure.rb

```diff
@@ -0,0 +1,444 @@
+      return {}
+    end
+
+    if zone_names_json.empty?
+      runner.registerError("Error in the #{user_hvac_json_path}. No zones specified under 'systems'.[].'thermal_zones'")
+      return {}
+    end
+
+    # Check for duplicate zones. First create a hash of every zones count
+    count_occurrences = zone_names_json.each_with_object(Hash.new(0)) { |zone, counts| counts[zone] += 1 }
+
+    # Select and return elements that have a count greater than 1 (i.e., duplicates)
+    duplicates = count_occurrences.select { |zone, count| count > 1 }.keys
+
+    unless duplicates.empty?
+      runner.registerError("Error in the #{user_hvac_json_path}. Duplicate zones found: #{duplicates.join(', ')}")
+      return {}
+    end
+
+    # Check the zones in the hash against those in the OS model
+    zone_names_os_model = []
+    model.getThermalZones.each do |zone|
+      zone_names_os_model << zone.name.to_s
+    end
+
+    # Check for zones in JSON that don't exist in OS model using array subtraction
+    nonexistant_zones = zone_names_json - zone_names_os_model
+
+    unless nonexistant_zones.empty?
+      runner.registerError("Error in the #{user_hvac_json_path}. The following zones don't exist in building "\
+                           "'#{model.building.get.name.to_s}': #{nonexistant_zones.join(', ')}. NOTE, zone names are "\
+                           "case sensitive.")
+      return {}
+    end
+
+    runner.registerInfo("No issues found in: #{user_hvac_json_path}.")
+    return hvac_mapping_hash
+
+  end
+
+  module CreateTypicalBldgConstants
+
+    CLIMATE_ZONES = ['Lookup From Model',  'ASHRAE 169-2013-1A', 'ASHRAE 169-2013-2A', 'ASHRAE 169-2013-2B',
+                     'ASHRAE 169-2013-3A', 'ASHRAE 169-2013-3B', 'ASHRAE 169-2013-3C', 'ASHRAE 169-2013-4A',
+                     'ASHRAE 169-2013-4B', 'ASHRAE 169-2013-4C', 'ASHRAE 169-2013-5A', 'ASHRAE 169-2013-5B',
+                     'ASHRAE 169-2013-6A', 'ASHRAE 169-2013-6B', 'ASHRAE 169-2013-7A', 'ASHRAE 169-2013-8A']
+
+    GEOMETRY_FILES = ['Existing Geometry',
+                      'ASHRAESmallOffice.osm',
+                      'ASHRAESmallHotel.osm',
+                      'ASHRAESecondarySchool.osm',
+                      'ASHRAERetailStripmall.osm',
+                      'ASHRAEQuickServiceRestaurant.osm',
+                      'ASHRAEPrimarySchool.osm',
+                      'ASHRAEOutpatient.osm',
+                      'ASHRAEMidriseApartment.osm',
+                      'ASHRAEMediumOffice.osm',
+                      'ASHRAELargeOffice.osm',
+                      'ASHRAELargeHotel.osm',
+                      'ASHRAELaboratory.osm',
+                      'ASHRAEHospital.osm',
+                      'ASHRAEHighriseApartment.osm',
+                      'ASHRAEFullServiceRestaurant.osm',
+                      'ASHRAECourthouse.osm',
+                      'ASHRAECollege.osm',
+
+    ]
+
+    HVAC_TYPES = ['Inferred',
```

Comment:
@jugonzal07 - Weren't we going to push these to a module?

Reply:
Fixed and update tested on OS 1.7 Application. Worked just fine.